src/uct/ib/rdmacm/rdmacm_iface.c

```diff
@@ -244,21 +282,25 @@ static void uct_rdmacm_iface_process_event(uct_rdmacm_iface_t *iface, struct rdm
                   ucs_sockaddr_str(remote_addr, ip_port_str, UCS_SOCKADDR_STRING_LEN));
 
         if (!iface->is_server) {
-            uct_rdmacm_client_handle_failure(iface);
+            destroy_cm_id = 1;
+            uct_rdmacm_client_handle_failure(iface, ep);
         }
         break;
 
     default:
         ucs_warn("unexpected RDMACM event: %d", event->event);
         break;
     }
+
+    return destroy_cm_id;
 }
 
 static void uct_rdmacm_iface_event_handler(int fd, void *arg)
 {
     uct_rdmacm_iface_t *iface = arg;
     struct rdma_cm_event *event;
-    int ret;
+    int ret, destroy_cm_id;
+    uct_rdmacm_ctx_t *rdmacm_ctx = NULL;
```

Comment:
pls use `uct_rdmacm_ctx_t *cm_id_ctx` in all places to be consistent with uct_rdmacm_ep